xhistogram/test/test_core.py

```diff
@@ -70,6 +96,30 @@ def test_histogram_results_2d():
     np.testing.assert_array_equal(hist, h)
 
 
+def test_histogram_results_2d_density():
+    nrows, ncols = 5, 20
+    data_a = np.random.randn(nrows, ncols)
+    data_b = np.random.randn(nrows, ncols)
+    nbins_a = 9
+    bins_a = np.linspace(-4, 4, nbins_a + 1)
+    nbins_b = 10
+    bins_b = np.linspace(-4, 4, nbins_b + 1)
+
+    h = histogram(data_a, data_b, bins=[bins_a, bins_b], density=True)
+    assert h.shape == (nbins_a, nbins_b)
+
+    hist, _, _ = np.histogram2d(data_a.ravel(), data_b.ravel(),
+                                bins=[bins_a, bins_b], density=True)
+    np.testing.assert_allclose(hist, h)
+
+    # check integral is 1
+    widths_a = np.diff(bins_a)
+    widths_b = np.diff(bins_b)
+    areas = np.outer(widths_a, widths_b)
+    integral = np.sum(hist * areas)
+    np.testing.assert_allclose(integral, 1.0)
```

Comment:
This is fine. We never expect exact numerical equivalence with floating point precision.